### PR TITLE
Add "Get started" tutorial

### LINKED ISSUE #10315: write a "get started with Qiskit" tutorial
Part of https://github.com/Qiskit/qiskit-tutorials/issues/1473

The [qiskit intro](https://github.com/Qiskit/qiskit-metapackage/blob/master/docs/intro_tutorial1.rst), [circuit basics](https://github.com/Qiskit/qiskit-tutorials/blob/master/tutorials/circuits/01_circuit_basics.ipynb) and [getting started with qiskit](https://github.com/Qiskit/qiskit-tutorials/blob/master/tutorials/circuits/1_getting_started_with_qiskit.ipynb) tutorials need to be merged in a single "get started with Qiskit" tutorial in this repo.

Related issues:
 * https://github.com/Qiskit/qiskit-terra/issues/10613
 * https://github.com/Qiskit/qiskit-metapackage/issues/1497
 * https://github.com/Qiskit/qiskit-metapackage/issues/1352
 * https://github.com/Qiskit/qiskit-tutorials/issues/1434

### LINKED ISSUE #10613: Review and refactor docs intro pages
I'm creating this issue here for now as the files in question still live in this repo, but we can move this issue to terra later (or maybe consider this work as part of the migration)

In recent discussions with users it has been highlighted that there is some confusion around the "getting started", "qiskit introduction" and "qc in a nutshell" pages. It is unclear to users where they should go for certain needs. I propose the following changes, further ideas/discussion are very welcome!

**1. remove the "qc in a nutshell" page**
This might be a controversial suggestion, but I think in qiskit.org we should be moving away from trying to teach people quantum computing (at least not at this basic level) and instead be nudging people to go to the qiskit textbook for this kind of info.

**2. Rename "getting started" to "Installation"**
It was brought up in user feedback sessions that new users assumed that links to the getting started page would take them to a "hello world" type of example, or the "qc in a nutshell" type of content. I think using a term like "Installation" would be better for this page. We should also update the "next steps" section at the bottom depending on what we decide to do with the other pages

**3. rename "qiskit introduction" to "Get started" or "get started with qiskit" or similar.** 
The "qiskit introduction" page feels like the closest to a "hello world" type of example that should live under the getting started label. Also explicitly titling it "get started with qiskit" could help people understand that this is specifically for qiskit and not for learning quantm computing in general

**4. refactor the "qiskit introduction" page to better address new user needs.**
This probably requires some more thought/discussion, but we could also refer to the qiskit.org homepage and use some of the quick start examples from there and expand them a bit, including the primitives as well. We should also add links at the top to direct people to the installation page if they haven't already got qiskit installed and another link to the qiskit textbook if they need info on learning quantum computing

note: if we rename the getting started page we should update the qiskit.org landing page accordingly

### PR DESCRIPTION
**Pull request by frankharkins**
### Summary

Adds a new "Get started with Qiskit" tutorial to replace the three existing introduction tutorials.

### Details and comments

<details>
  <summary>Topics breakdown</summary>

Here's the topics I could see covered in the existing tutorials. Strikethough means I chose not to include it in the new tutorial.

- **General Qiskit workflow**
  - Build circuit
  - ~~Compile for device~~
  - Run
  - ~~Analyze results~~ (although I do show `plot_histogram`)

- **QuantumCircuit**
  - `h`
  - `cx`
  - `measure`
  - `draw`

- **Simulating**
  - Sampler
  - ~~`quantum_info.Statevector`~~
  - Qiskit Aer
  - ~~`AerSimulator`~~
  - ~~state vector vs qasm vs other simulators~~
  - `.result`
  - ~~`.get_counts` / `.get_memory`~~

- **Other important classes / functions**
  - `plot_histogram`
  - ~~`shots` parameter~~
  - ~~`transpile`~~
  - ~~BasicAer~~

- ~~**Settings**~~
  - ~~Change default circuit drawer~~

- ~~**General quantum computing**~~
  - ~~Default state of qubits~~
  - ~~Qubit ordering~~
  - ~~What a circuit is~~
</details>

### ⚠️ Note to reviewers

This notebook contains [raw reST cells](https://nbsphinx.readthedocs.io/en/0.9.2/raw-cells.html), which don't display in GitHub's built-in notebook viewer. For example [this cell](https://github.com/Qiskit/qiskit-terra/blob/7752e28366efcb51ef6c2f83b64e2fb26ac17993/docs/tutorials/get_started.ipynb#L18-L26) is hidden. This could be a downside of using notebooks with ReST cells.

***

Closes #10315, closes #10613

### CODE REVIEW

**Review by javabster — COMMENTED** [review]
Thanks for the speedy work on this @frankharkins!

The main general comment I have is that I think there could be a smidge more detail. Atm this feels more like a how to guide than a tutorial (not sure if anyone's showed you this yet but we have a docs guide [here](https://qiskit.github.io/qiskit_sphinx_theme/index.html) that breaks down the difference). I know we don't want to go too much down the rabbit hole of teaching quantum computing with these tutorials but I think we could give a bit more detail than whats currently here. One example of a tutorial that I think strikes the right balance is [this one](https://qiskit.org/ecosystem/ibm-runtime/tutorials/qpe_with_sampler.html) on QPE from the runtime docs. I think something similar here, giving some context on what the bell state actually is, would be good. Also would be good to add a pre-requisites section that links people to the installation page (see tutorial guidelines [here])(https://qiskit.github.io/qiskit_sphinx_theme/tutorials/tutorials_guidelines.html))

File: docs/tutorials/get_started.ipynb
Comment (frankharkins) [inline docs/tutorials/get_started.ipynb:35]: This CHSH link is temporary, just so you can see what it'll look like. I'll update the link once we have the page in production.

File: docs/tutorials/get_started.ipynb
Comment (frankharkins) [inline docs/tutorials/get_started.ipynb:190]: On the fence on whether to include this.

**Review by frankharkins — COMMENTED** [review]
Sorry for the delay. I've added a bit more detail, plus some links out for further explanation.

File: docs/tutorials/get_started.ipynb
Comment (javabster) [inline docs/tutorials/get_started.ipynb:100]: when mentioning the real quantum computer option could you explicitly mention "IBM Qiskit Runtime" and link to their docs?

File: docs/tutorials/get_started.ipynb
Comment (javabster) [inline docs/tutorials/get_started.ipynb:142]: this line feels very final but then there is still another section about plotting results so it feels a bit disjointed. Could you move this line to the end of the notebook, maybe before the qiskit community section?

File: docs/tutorials/get_started.ipynb
Comment (javabster) [inline docs/tutorials/get_started.ipynb:35]: It feels a bit weird jumping from the quantum circuit sentence into an explanation of entanglement. Is there  a way to make the transition feel a bit smoother? Maybe just an additional sentence at the start could help, or reordering the current information like "The quantum program you will be creating produces a bell state, often used to start entanglement experiments, entanglement is important property etc etc". Also might be worth starting with this entanglement section before the explanation of the quantum circuit class

**Review by javabster — COMMENTED** [review]
making good progress!

**Comment by javabster** [comment]
one additional point, the [existing getting started tutoria](https://qiskit.org/documentation/tutorials/circuits/1_getting_started_with_qiskit.html)l gives a helpful tip about how the qubit oredering is reversed, I think it would be good to do something similar here, or possibly just mention it briefly and then share a link to the more detailed [explanation](https://github.com/Qiskit/qiskit-terra/blob/main/docs/explanation/endianness.rst) (not deployed yet, also there is an [open PR/discussion ](https://github.com/Qiskit/qiskit-terra/pull/10205) on this that you might like to take over or weigh in on)

File: docs/tutorials/get_started.ipynb
Comment (javabster) [inline docs/tutorials/get_started.ipynb:187]:
```suggestion
    "And that's it! You now have everything you need to get going writing and running quantum programs."
```
A little less of a bold claim

Comment (Eric-Arellano) [reply docs/tutorials/get_started.ipynb:187]: How about "to start writing and running quantum programs"? My friends who are English language learners hate phrasal verbs and I suspect that "get going" might be less accessible.

File: docs/tutorials/get_started.ipynb
Comment (frankharkins) [reply docs/tutorials/get_started.ipynb:141]: I agree, thanks!

File: docs/tutorials/get_started.ipynb
Comment (frankharkins) [reply docs/tutorials/get_started.ipynb:187]: Quick case for the original: It's more direct, optimistic, and I don't think it's untrue?

File: docs/tutorials/get_started.ipynb
Comment (abbycross) [inline docs/tutorials/get_started.ipynb:139]:
```suggestion
    "The `quasi_dists` property is a list of sampled distributions for each circuit we provided. Since we only provided one circuit, the list only has one element.\n",
```

File: docs/tutorials/get_started.ipynb
Comment (abbycross) [inline docs/tutorials/get_started.ipynb:12]:
```suggestion
    "In this tutorial, you'll create a simple quantum program and run it on a simulator. Before you start, you'll need some way of running Qiskit code. See [Installation](https://qiskit.org/documentation/getting_started.html) to get set up, and be sure to install the recommended `qiskit[visualization]` support.\n",
```

Comment (frankharkins) [reply docs/tutorials/get_started.ipynb]: Good idea, thanks!

File: docs/tutorials/get_started_with_qiskit.ipynb
Comment (Eric-Arellano) [inline docs/tutorials/get_started_with_qiskit.ipynb:126]: This link might be outdated for Qiskit Runtime. Maybe sync with the relevant time on what link you should use.

File: docs/tutorials/get_started_with_qiskit.ipynb
Comment (Eric-Arellano) [inline docs/tutorials/get_started_with_qiskit.ipynb:189]: Maybe link to https://qiskit.org/documentation/tutorials/circuits/2_plotting_data_in_qiskit.html for more info on visualization.

File: docs/tutorials/get_started_with_qiskit.ipynb
Comment (Eric-Arellano) [inline docs/tutorials/get_started_with_qiskit.ipynb:225]: Would it make sense to discuss where to go next? Maybe simply linking to the tutorials index at https://qiskit.org/documentation/tutorials/index.html?

File: docs/tutorials/get_started_with_qiskit.ipynb
Comment (Eric-Arellano) [inline docs/tutorials/get_started_with_qiskit.ipynb:259]: In the rendered output, the versions are out of date. I think the notebook needs regenerating. FYI CI will upload an artifact to Azure with the executed notebooks, run in a standard environment.

**Review by Eric-Arellano — COMMENTED** [review]
Looks good! As discussed, this either needs to live in the top level of `tutorials/` and be added to `index.rst`, or move into a subfolder of `tutorials/`.

Also this PR should remove the now irrelevant tutorials.

File: docs/tutorials/get_started_with_qiskit.ipynb
Comment (frankharkins) [reply docs/tutorials/get_started_with_qiskit.ipynb]: You mean https://qiskit.org/documentation/tutorials.html?

Comment (Eric-Arellano) [reply docs/tutorials/get_started_with_qiskit.ipynb]: Ah yeah, I did. Thanks.

File: docs/tutorials/get_started_with_qiskit.ipynb
Comment (frankharkins) [reply docs/tutorials/get_started_with_qiskit.ipynb]: Good call, updated

File: docs/tutorials/get_started_with_qiskit.ipynb
Comment (frankharkins) [reply docs/tutorials/get_started_with_qiskit.ipynb]: I couldn't find my notebook in the `executed_tutorials` zip, but I've re-run with the latest version of Qiskit.

**Review by javabster — DISMISSED** [review]
LGTM 🚀

File: docs/installation.rst
Comment (Eric-Arellano) [inline docs/installation.rst]: Please don't merge yet. I realized this will break the top nav bar: 

Screenshot 2023-08-21 at 7 50 02 AM

We definitely want to go ahead with renaming the page title to Installation, but the HTML page needs to still be `getting_started.html` for now. You can do that by restoring the file name, but still changing the page title like you do here.

I think the only way for a graceful migration here would be to set up a redirect. But given that the docs will change soon to a new structure, I'm not sure it matters enough to rename this page. I imagine we have several third-party links to this page, so probably simplest to keep it this name for now.

**Review by Eric-Arellano — CHANGES_REQUESTED** [review]

File: docs/installation.rst
Comment (frankharkins) [reply docs/installation.rst]: Great spot – should be fixed in [7f85048](https://github.com/Qiskit/qiskit/pull/10328/commits/7f85048b30858dcfe2881cf2049bc865c13e012b)

**Review by Eric-Arellano — DISMISSED** [review]

File: docs/get_started_with_qiskit.ipynb
Comment (mtreinish) [inline docs/get_started_with_qiskit.ipynb:18]: Does a relative link like this to the rst file get resolved correctly by `nbsphinx`?

File: docs/get_started_with_qiskit.ipynb
Comment (mtreinish) [inline docs/get_started_with_qiskit.ipynb]: The normal capitalization of matplotlib is `Matplotlib`: https://matplotlib.org/

File: docs/get_started_with_qiskit.ipynb
Comment (mtreinish) [inline docs/get_started_with_qiskit.ipynb]: Do you want to explain that outside of jupyter you'll have to manually handle file output or display of the images?

File: docs/get_started_with_qiskit.ipynb
Comment (mtreinish) [inline docs/get_started_with_qiskit.ipynb]: I would maybe avoid saying "random results" here as it kind of implies a random distribution. Typically I've seen this expressed as "non-deterministic results" or "probabilistic" in intro to QC guides. Like in this case you're expecting a 50/50 distribution of `00` or `11` for most people's intuition they wouldn't call that random.

File: docs/get_started_with_qiskit.ipynb
Comment (mtreinish) [inline docs/get_started_with_qiskit.ipynb]: Would it be better to link to the aer doc's sampler page here? https://qiskit.org/ecosystem/aer/stubs/qiskit_aer.primitives.Sampler.html#qiskit_aer.primitives.Sampler

File: docs/get_started_with_qiskit.ipynb
Comment (mtreinish) [inline docs/get_started_with_qiskit.ipynb]: If we're in rst space can we use references instead of external links? If we can that'll be a bit more robust for different builds of the documentation.

**Review by mtreinish — COMMENTED** [review]
This looks fine to me, I'm not super keen on using a notebook for this, it makes it a lot harder to review and interact with but that's not a strong objection. I left a few comments inline about the content but nothing major.

The other question is about the "Quantum computing in a nutshell" guide, that is being removed here, but this new document isn't a replacement for that information; it only replaces the intro tutorial (and moves to using the sampler). Is there a plan to replace that guide with some of the higher level background info about quantum computing?

File: docs/get_started_with_qiskit.ipynb
Comment (frankharkins) [reply docs/get_started_with_qiskit.ipynb:18]: Yep! Here's from my build

Screenshot 2023-08-29 at 12 25 42

File: docs/get_started_with_qiskit.ipynb
Comment (frankharkins) [reply docs/get_started_with_qiskit.ipynb]: Good idea, I've taken another pass at it

File: docs/get_started_with_qiskit.ipynb
Comment (frankharkins) [reply docs/get_started_with_qiskit.ipynb]: Good point, thanks!

File: docs/get_started_with_qiskit.ipynb
Comment (frankharkins) [reply docs/get_started_with_qiskit.ipynb]: I don't think this is possible until #10710 is merged

**Comment by frankharkins** [comment]
@mtreinish thank you!

> I'm not super keen on using a notebook for this, it makes it a lot harder to review and interact with but that's not a strong objection.

I also don't have a strong opinion; I did ask which format to use before writing and the answers were pretty evenly divided.

> The other question is about the "Quantum computing in a nutshell" guide, [...] Is there a plan to replace that guide with some of the higher level background info about quantum computing?

I believe the plan is not to replace this, instead pointing people to the learning platform ([source](https://github.com/Qiskit/qiskit/issues/10613#issue-1847373755)).

**Pull request closed without merging**
